First Spring Boot Project

obedjamir opened the pull request:
First spring boot project

obedjamir committed:
First Spring Boot Project

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/TestController.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @GetMapping("/API2")
+    public String getAPi2()
```
Add a queryParam to pass data from the user

obedjamir replied:
@RequestParam used

PinakiGhosh commented on pom.xml:
```diff
@@ -0,0 +1,51 @@
+			<artifactId>spring-boot-starter-actuator</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.data</groupId>
```
This is not required

obedjamir replied:
Dependencies removed

obedjamir committed:
First Spring Boot Project
@RequestParam added
dependencies removed

obedjamir committed:
First Spring Boot Project
Used lombok library

obedjamir committed:
First Spring Boot Project
Spring Data JPA used to connect to mysql database
Two tables created with foreign key relation
CURD controller implemented for tables

obedjamir committed:
First Spring Boot Project
Foreign key changed to Many to one relationship
implementation of spring security

obedjamir committed:
First Spring Boot Project
removed unnecessary import

obedjamir committed:
First Spring Boot Project
employee count in Department is updated with every new Employee
Admin user set up with username and password

PinakiGhosh commented on pom.xml:
```diff
@@ -0,0 +1,66 @@
+		<dependency>
+			<groupId>com.google.code.gson</groupId>
+			<artifactId>gson</artifactId>
+			<version>2.9.0</version>
```
Add the raw versions as properties

obedjamir replied:
This issues has been resolved.

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -0,0 +1,49 @@
+
+import java.util.List;
+
+@RestController
```
The url mapping should be like
@RestController
@RequestMapping("/departments")

Then the individual functions should just be the method. Ex POST on this url will signify create, get will signify fetch etc..

obedjamir replied:
I have implemented @RequestMapping as requested

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/EmployeeController.java:
```diff
@@ -0,0 +1,51 @@
+
+import java.util.List;
+
+@RestController
```
Same as previous conntroller

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -0,0 +1,49 @@
+        return service.saveDepartment(department);
+    }
+
+    @PostMapping("/addDepartmentList")
```
No need to have api for multiple insert together

obedjamir replied:
I have reoved this api

PinakiGhosh commented on src/main/java/com/bankbazaar/test/entity/Department.java:
```diff
@@ -0,0 +1,24 @@
+    @Id
+    @GeneratedValue
+    @Column(name="department_id")
+    private int id;
```
Have id as Long

obedjamir replied:
id is now long type

PinakiGhosh commented on src/main/java/com/bankbazaar/test/entity/Employee.java:
```diff
@@ -0,0 +1,30 @@
+    @Id
+    @GeneratedValue
+    @Column(name="id")
+    private int id;
```
Have the id as Long

PinakiGhosh commented on src/main/java/com/bankbazaar/test/entity/Department.java:
```diff
@@ -0,0 +1,24 @@
+    private int id;
+    @Column(name="department_name")
+    private String name;
+    @Column(name="employee_count")
```
Ideally department should not have this. This should be derived from the employee table

obedjamir replied:
I have replaced employeeCount with departmentFloor and departmentPhone

PinakiGhosh commented on src/main/java/com/bankbazaar/test/service/EmployeeService.java:
```diff
@@ -0,0 +1,117 @@
+
+            ResponseEntity<String> response = new RestTemplate().exchange(uriGet, HttpMethod.GET, request, String.class);
+
+            JsonObject departmentData = JsonParser.parseString(response.getBody()).getAsJsonObject();
```
Is this for testing?

obedjamir replied:
Yes, i was trying to work with external apis.

PinakiGhosh commented on src/main/java/com/bankbazaar/test/service/EmployeeService.java:
```diff
@@ -0,0 +1,117 @@
+        }
+        catch (Exception e)
+        {
+            System.out.println("The provided department ID doesn't exist");
```
Dont use System.out.println in projects. Always use logs.
Look into @Slf4j annotation

obedjamir replied:
okay. however i have removed this part of the code for now.

PinakiGhosh commented on src/main/java/com/bankbazaar/test/entity/Department.java:
```diff
@@ -0,0 +1,24 @@
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+@Table(name = "Department")
```
Table names should begin with a lower case letter

obedjamir replied:
THis issue has been resolved.

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -0,0 +1,49 @@
+        return service.getDepartmentList();
+    }
+
+    @GetMapping("/departmentById/{id}")
```
This should return a 404 response when nothing is found. Implement it for other get apis as well.

Use ResponseEntity for the response

obedjamir replied:
Necessary changes have been made to implement ResponseEntity

obedjamir committed:
First Spring Boot Project
All requested changes implemented

PinakiGhosh commented on pom.xml:
```diff
@@ -36,7 +38,6 @@
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-data-jpa</artifactId>
-			<version>2.6.5</version>
```
Is the project building like this?
You have to give <version>${gson.version}</version>

obedjamir replied:
Changes have been made

PinakiGhosh commented on pom.xml:
```diff
@@ -36,7 +38,6 @@
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-data-jpa</artifactId>
-			<version>2.6.5</version>
```
<version>${gson.version}</version>

PinakiGhosh replied:
mvn clean install

obedjamir replied:
Required changes have been made

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -3,47 +3,102 @@
-import java.util.List;
 
 @RestController
+@RequestMapping("/department")
```
departments

obedjamir replied:
CHanged to departments

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -3,47 +3,102 @@
-    public Department addDepartment(@RequestBody Department department) {
-        return service.saveDepartment(department);
-    }
+    @RequestMapping(value = "/post", method = RequestMethod.POST)
```
POST /departments -> Create
GET /departmentd/id -> Fetch
DELETE /departments/id -> Delete

obedjamir replied:
This issue has been resolved

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -3,47 +3,102 @@
-        return service.saveDepartment(department);
-    }
+    @RequestMapping(value = "/post", method = RequestMethod.POST)
+    public ResponseEntity<String> addDepartment(@RequestBody Department department) {
```
This should be Department

obedjamir replied:
Changed to Department type

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -3,47 +3,102 @@
+        }
+        catch(Exception e)
+        {
+            return new ResponseEntity<>(
```
This is not a not found scenario

obedjamir replied:
This issue has been resolved

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -3,47 +3,102 @@
-        return service.saveDepartment(department);
-    }
+    @RequestMapping(value = "/post", method = RequestMethod.POST)
+    public ResponseEntity<String> addDepartment(@RequestBody Department department) {
```
Add validations to request body

obedjamir replied:
@Valid used for validation

PinakiGhosh commented on src/main/java/com/bankbazaar/test/controller/DepartmentController.java:
```diff
@@ -3,47 +3,102 @@
+    @RequestMapping(value = "/get/id/{id}", method = RequestMethod.GET)
+    public ResponseEntity<String> findDepartmentById(@PathVariable Long id) {
+
+        if (service.getDepartmentById(id).isEmpty()) {
```
Dont query the DB twice

obedjamir replied:
This issue has been resolved

PinakiGhosh commented on src/main/java/com/bankbazaar/test/entity/Department.java:
```diff
@@ -11,14 +11,16 @@
     @GeneratedValue
     @Column(name="department_id")
-    private int id;
+    private long id;
```
Long, Integer

obedjamir replied:
changes have been made

PinakiGhosh commented on src/main/java/com/bankbazaar/test/service/DepartmentService.java:
```diff
@@ -50,8 +43,14 @@ public Department getDepartmentByName(String name) {
-    public String deleteDepartment(int id) {
+    public String deleteDepartment(long id) {
+        Department existingDepartment = departmentRepository.findById(id).orElse(null);
+        if(existingDepartment==null)
```
if (departmentRepository.findById(id).isEmpty())

obedjamir replied:
i haven't made changes to this part as "existingDepartment" object is used for updating Department

obedjamir committed:
First Spring Boot Project
All requested changes have been made
